## osu! fix: patch from server-default-integrity patchset from wine-staging removed in commit
## 9e265ac738bfd89b50071e9d0d881fe97f652c16 (9.4) needed to fix osu:// protocol links crashing in c0000135.

From 643461bc989bd848363241dcfa04187e8d3a84d1 Mon Sep 17 00:00:00 2001
From: Zebediah Figura <z.figura12@gmail.com>
Date: Fri, 21 May 2021 21:52:06 -0500
Subject: [PATCH] ntdll: Always start the initial process through start.exe.

Signed-off-by: Zebediah Figura <z.figura12@gmail.com>
---
 dlls/ntdll/unix/env.c | 26 +++-----------------------
 1 file changed, 3 insertions(+), 23 deletions(-)

diff --git a/dlls/ntdll/unix/env.c b/dlls/ntdll/unix/env.c
index 30782a70eb0..959bafb28b7 100644
--- a/dlls/ntdll/unix/env.c
+++ b/dlls/ntdll/unix/env.c
@@ -1909,6 +1909,7 @@ static void init_peb( RTL_USER_PROCESS_PARAMETERS *params, void *module )
  */
 static RTL_USER_PROCESS_PARAMETERS *build_initial_params( void **module )
 {
+    static const char *args[] = { "start.exe", "/exec" };
     static const WCHAR valueW[] = {'1',0};
     static const WCHAR pathW[] = {'P','A','T','H'};
     RTL_USER_PROCESS_PARAMETERS *params = NULL;
@@ -1937,29 +1938,8 @@ static RTL_USER_PROCESS_PARAMETERS *build_initial_params( void **module )
     add_registry_environment( &env, &env_pos, &env_size );
     env[env_pos++] = 0;
 
-    status = load_main_exe( NULL, main_argv[1], curdir, 0, &image, module );
-    if (!status)
-    {
-        char *loader;
-
-        if (main_image_info.ImageCharacteristics & IMAGE_FILE_DLL) status = STATUS_INVALID_IMAGE_FORMAT;
-        /* if we have to use a different loader, fall back to start.exe */
-        if ((loader = get_alternate_wineloader( main_image_info.Machine )))
-        {
-            free( loader );
-            status = STATUS_INVALID_IMAGE_FORMAT;
-        }
-    }
-
-    if (status)  /* try launching it through start.exe */
-    {
-        static const char *args[] = { "start.exe", "/exec" };
-        free( image );
-        if (*module) NtUnmapViewOfSection( GetCurrentProcess(), *module );
-        load_start_exe( &image, module );
-        prepend_argv( args, 2 );
-    }
-    else rebuild_argv();
+    load_start_exe( &image, module );
+    prepend_argv( args, 2 );
 
     main_wargv = build_wargv( get_dos_path( image ));
     cmdline = build_command_line( main_wargv );
-- 
2.40.1

